lib/explorer/series.ex

```diff
@@ -662,25 +662,17 @@ defmodule Explorer.Series do
       iex> Explorer.Series.quantile(s, 0.5)
       2.0
 
-      iex> s = Explorer.Series.from_list([~D[2021-01-01], ~D[1999-12-31]])
-      iex> Explorer.Series.quantile(s, 0.5)
-      ~D[2021-01-01]
-
-      iex> s = Explorer.Series.from_list([~N[2021-01-01 00:00:00], ~N[1999-12-31 00:00:00]])
-      iex> Explorer.Series.quantile(s, 0.5)
-      ~N[2021-01-01 00:00:00.000]
-
       iex> s = Explorer.Series.from_list([true, false, true])
       iex> Explorer.Series.quantile(s, 0.5)
-      ** (ArgumentError) Explorer.Series.quantile/2 not implemented for dtype :boolean. Valid dtypes are [:integer, :float, :date, :datetime].
+      ** (ArgumentError) Explorer.Series.quantile/2 not implemented for dtype :boolean. Valid dtypes are [:integer, :float].
   """
   @spec quantile(series :: Series.t(), quantile :: float()) :: any()
   def quantile(%Series{dtype: dtype} = series, quantile)
-      when dtype in [:integer, :float, :date, :datetime],
+      when dtype in [:integer, :float],
```

Comment:
Should we still support date and datetime by casting and then converting back to date/datetime? Do you know what is the rationale for them to no longer support date/datetime?

Reply:
In any case, if we are no longer supporting this, the docs have to be updated. :)

Reply:
I couldn't find the reason why they don't work anymore :/

The only problem I see trying to casting to integers and converting is that maybe the value is not a valid datetime / date anymore. I could return `nil` for invalid cases. WDYT?

@cigrainger thoughts about this? I will update the docs if you agree to remove support for this.

Reply:
The datetime and date are contiguous, so we should be fine!

Reply:
This is where I was getting stuck as well. I couldn't find a rationale for dropping it but there's been a fair amount of reworking. `polars` is definitely moving fast and breaking things. I'm fine with dropping this for now and revisiting it at a later date.

Reply:
I was able to make the casting work. It should have the same behavior as before :)